unlock-app/src/services/walletService.js

```diff
@@ -26,7 +28,7 @@ export const keyId = (lock, owner) => [lock, owner].join('-')
  * actually retrieving the data from the chain/smart contracts
  */
 export default class WalletService extends EventEmitter {
-  constructor(availableProviders = providers) {
+  constructor(availableProviders, runningOnServer, wait = delayPromise) {
```

Comment:
Still not a huge fan of the `wait` thing here, especially because I worry this will create weird problems when we add params to this function.
What do you think of passing a object like this:
```suggestion
  constructor({availableProviders: config.providers, runningOnServer: config.isServer, runningOnServer, wait:  delayPromise} = configure()) {
```

This lets us just call this with no object and would use the config by default, making "pure" function while still avoiding the hassle of drilling down the config all over?

Reply:
Sure so perhaps a new PR to include the promise into the configuration? Then we extract it like this?

Reply:
Sure so perhaps a new PR to include the promise into the configuration? Then we extract it like this?

Reply:
Something feels odd about including a promise-that-never-changes in configuration. Perhaps a 2nd parameter for utils so we get:

```js
constructor({availableProviders: config.providers, runningOnServer: config.isServer, runningOnServer}, { delayPromise }}) {
```

Also, I removed the default value of `configure()` from my example, primarily because we can easily do that inside `walletMiddleware`, which is currently the only place `walletService` is initialized in any case. What do you think?